### ISSUE TITLE
Regression with `:clj-kondo/ignore [...]` in hooks

### ISSUE THREAD

**Issue by tonsky** [clj-kondo/clj-kondo#2404]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2024.09.26

**platform**

native mac m1

**editor**

cli

**problem**

Running `clj-kondo --lint`, I get:

```
Unexpected error. Please report an issue.
java.lang.IllegalArgumentException: Key must be integer
	at clojure.lang.APersistentVector.assoc(APersistentVector.java:347)
	at clojure.lang.APersistentVector.assoc(APersistentVector.java:18)
	at clojure.lang.RT.assoc(RT.java:827)
	at clojure.core$assoc__5434.invokeStatic(core.clj:195)
	at clj_kondo.impl.linters$lint_redundant_ignores.invokeStatic(linters.clj:865)
	at clj_kondo.core$run_BANG_.invokeStatic(core.clj:230)
	at clj_kondo.main$main.invokeStatic(main.clj:139)
	at clj_kondo.main$main.doInvoke(main.clj:135)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clj_kondo.main$_main$fn__14215.invoke(main.clj:186)
	at clj_kondo.main$_main.invokeStatic(main.clj:186)
	at clj_kondo.main$_main.doInvoke(main.clj:184)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj_kondo.main.main(Unknown Source)
```

Seems related to this hook:

```
(ns hooks.io.pedestal.log
  (:require [clj-kondo.hooks-api :as api]))

(defn log-expr
  "Expands (log-expr :a :A :b :B ... )
  to (hash-map :a :A :b :B ... ) per clj-kondo examples."
  [{:keys [:node]}]
  (let [[k v & _kvs] (rest (:children node))]
    (when-not (and k v)
      (throw (ex-info "No kv pair provided" {})))
    (let [new-node (api/list-node
                    (list*
                     (api/token-node 'hash-map)
                     (rest (:children node))))]
      {:node (vary-meta new-node assoc :clj-kondo/ignore [:unused-value])})))
```

If I comment out `vary-meta`, it works

Used to work on 2024.08.29

**Comment by borkdude** [comment]
Thanks, I'll take a look
